packages/x-tree-view/src/internals/plugins/useTreeViewContextValueBuilder/useTreeViewContextValueBuilder.ts

```diff
@@ -0,0 +1,27 @@
+import useId from '@mui/utils/useId';
+import { TreeViewInstance, TreeViewPlugin } from '../../models';
+import { UseTreeViewContextValueBuilderSignature } from './useTreeViewContextValueBuilder.types';
+
+export const useTreeViewContextValueBuilder: TreeViewPlugin<
+  UseTreeViewContextValueBuilderSignature
+> = ({ instance, params }) => {
+  const treeId = useId(params.id);
+
+  return {
+    getRootProps: () => ({
+      id: treeId,
+    }),
+    contextValue: {
```

Comment:
Is it worth memoizing the context value?

Reply:
It was not memoized before and it would not bring any benefit for now
We can consider memoizing it in the future if it make sense :+1: